subprojects/reporting/src/main/java/org/gradle/api/reporting/Report.java

```diff
@@ -88,8 +89,11 @@ public String determineName(Report report) {
      * <p>
      * Subtypes may implement setters for the destination.
      *
+     * Subtypes need to add the correct output annotation, depending on whether the output of the report is a directory or a file.
```

Comment:
Why not add this message in the `@Internal`'s `value()`? This is not very useful information for whoever's reading the (inherited) Javadoc.

Reply:
Done. I forgot `@Internal` can have a value.